clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -727,6 +732,17 @@ public ConsumerRecords<K, V> poll(final Duration timeout) {
         }
     }
 
+    private boolean isGenerationKnown() {
```

Comment:
How about this one-liner

```
return groupMetadata.filter(g -> g.generationId() != JoinGroupRequest.UNKNOWN_GENERATION_ID).isPresent();
```

Reply:
I am not sure this is correct, because I can use user assigned partitions but still use a group ID to manage the offsets broker-side, if I remember correctly.

Reply:
Probably, I can do something like
```java
if (subscriptions.hasAutoAssignedPartitions()) {
    return groupMetadata.filter(g -> g.generationId() != JoinGroupRequest.UNKNOWN_GENERATION_ID).isPresent(); 
}
return true;
```

Reply:
Makes sense. But then I'm a bit confused by the method name, because it just returns true when there is no generation...